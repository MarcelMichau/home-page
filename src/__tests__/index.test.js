import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import IndexPage from '../pages/index';

const data = {
	markdownRemark: {
		html:
			'<p>This is my website. No one else\'s.</p>\n<p>Look how minimal it is.</p>\n<p>It\'s built using:</p>\n<ul>\n<li><a href="https://www.typescriptlang.org/">TypeScript</a></li>\n<li><a href="https://reactjs.org/">React</a></li>\n</ul>\n<p>Along with:</p>\n<ul>\n<li><a href="https://www.gatsbyjs.org/">Gatsby</a></li>\n</ul>\n<p>With some help from:</p>\n<ul>\n<li><a href="https://www.styled-components.com/">Styled Components</a></li>\n<li><a href="https://fontawesome.com/">Font Awesome</a></li>\n<li><a href="https://mattboldt.com/demos/typed-js/">Typed.js</a></li>\n<li><a href="https://getwaves.io/">getwaves.io</a></li>\n</ul>\n<p>With code kept in line by:</p>\n<ul>\n<li><a href="https://codeclimate.com/github/MarcelMichau/home-page">Code Climate</a></li>\n<li><a href="https://prettier.io/">Prettier</a></li>\n</ul>\n<p>Tested using:</p>\n<ul>\n<li><a href="https://jestjs.io/">Jest</a></li>\n<li><a href="https://www.docker.com/">Docker</a></li>\n<li><a href="https://travis-ci.com/MarcelMichau/home-page">Travis CI</a></li>\n</ul>\n<p>Deployed to:</p>\n<ul>\n<li><a href="https://vercel.com/">Vercel</a></li>\n</ul>\n<p>And code hosted on:</p>\n<ul>\n<li><a href="https://github.com/MarcelMichau/home-page">GitHub</a></li>\n</ul>\n<p>It is intended to be:</p>\n<ul>\n<li>Fast 🐌</li>\n<li>Agile 🐢</li>\n<li>Scalable ⚖️</li>\n<li>Cloud-native 🌧️</li>\n<li>Enterprise-ready 🏢</li>\n<li>Containerized 📦</li>\n<li>Microservice-based 🔍</li>\n<li>Powered by Machine Learning 🤖</li>\n<li>Blockchain ₿</li>\n<li>Filled with all the buzz words so that search engines pick this up 😁</li>\n</ul>',
	},
	file: {
		childImageSharp: {
			fixed: {
				base64:
					'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGQABAAMBAQAAAAAAAAAAAAAAAAMEBQEC/8QAFwEBAAMAAAAAAAAAAAAAAAAAAgABA//aAAwDAQACEAMQAAABn7XkD8JV55GmRVQD/8QAHRAAAgIBBQAAAAAAAAAAAAAAAQIAAyEEERITFP/aAAgBAQABBQInZEDz1QENp0GEqrVaXPbdicZ//8QAFhEBAQEAAAAAAAAAAAAAAAAAEAFB/9oACAEDAQE/AYYf/8QAFhEBAQEAAAAAAAAAAAAAAAAAEAEx/9oACAECAQE/AS6f/8QAHxAAAgECBwAAAAAAAAAAAAAAAAESESECEDFRYXGB/9oACAEBAAY/ApcEpeGg6jdWl2XaZh2ZBWTWX//EABsQAQACAwEBAAAAAAAAAAAAAAEAESFBUTFx/9oACAEBAAE/IR36EaWu+4Yg6lQwLlQNTFQQ7EgtpSahOxEPsw2z/9oADAMBAAIAAwAAABCzCP8A/8QAGBEBAQEBAQAAAAAAAAAAAAAAAQAhETH/2gAIAQMBAT8QIm2x7uF//8QAFhEBAQEAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/EKyoMf/EAB0QAQACAgMBAQAAAAAAAAAAAAEAESFRMUFh0aH/2gAIAQEAAT8QUksYHtoD9lbDmSttETkb4csBoBAX2JCGZVxBx77ATWqpCU1pTs63HAbfYU+RDQqc5r5P/9k=',
				width: 300,
				height: 300,
				src: '/static/ceeab084c21596435216ed7397fb861b/25e3f/profile.jpg',
				srcSet:
					'/static/ceeab084c21596435216ed7397fb861b/25e3f/profile.jpg 1x,\n/static/ceeab084c21596435216ed7397fb861b/f11e0/profile.jpg 1.5x,\n/static/ceeab084c21596435216ed7397fb861b/7b2a1/profile.jpg 2x',
				srcWebp: '/static/ceeab084c21596435216ed7397fb861b/50ead/profile.webp',
				srcSetWebp:
					'/static/ceeab084c21596435216ed7397fb861b/50ead/profile.webp 1x,\n/static/ceeab084c21596435216ed7397fb861b/27933/profile.webp 1.5x,\n/static/ceeab084c21596435216ed7397fb861b/ab687/profile.webp 2x',
			},
		},
	},
};

jest.mock('../components/TypingSubtitle', () => () => <></>);

const renderIndexPageToJSON = () => {
	return renderer.create(<IndexPage data={data} />).toJSON();
};

describe('IndexPage', () => {
	it('renders correctly', () => {
		expect(renderIndexPageToJSON()).toMatchSnapshot();
	});
});
