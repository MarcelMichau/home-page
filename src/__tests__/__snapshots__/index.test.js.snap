// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<div>
  <div>
    <div
      className="sc-bZQynM kbUix"
    >
      <header
        className="sc-gzVnrw jrHBcJ sc-bdVaJa hLZfwv"
      >
        <div
          className="sc-bwzfXH VcUjL"
        />
        <div
          className="sc-htpNat ihPAez"
        >
          <div>
            <h2>
              Marcel Michau
            </h2>
            <div
              className="sc-iwsKbI eRsqeS gatsby-image-wrapper"
              style={
                Object {
                  "display": "inline-block",
                  "height": 200,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 200,
                }
              }
            >
              <img
                alt=""
                src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='200' height='200' viewBox='0 0 200 200' version='1'%3e%3cpath d='M1 15a1211 1211 0 0 1 1 79c1 8 1 20-1 22l-1 43v41h75l-3-3-24-23c-3-4-3-4-1-6v-1l-1-1h2l2-1c0-2 6-1 6 1 3 12 7 13 11 5 2-4 6-6 11-3 9 3 36 6 34 4l-2-1h-3c-2 0-6-8-6-9l-2-4-3-4-2-4v-2l2-1c1-1 1-1 2 1 0 2 1 2 4-2l3-3c-1-1 0-1 3-1l16-1h-5l-5-1c-1-2-3-2-12-3-2 0-2 0-1-1l4-2c2-1 3-2 4-1l3-2c1-2 2-3 4-3s4-2 2-3c-3-1-4-5-3-10l1-8c-1-6 0-8 3-8 4-1 5 0 5 3 0 4 4 14 5 13l-2-5v-8l-1-1h3v1l8 2 2-3h1l1 2c0 1 3 1 9-1 2 0 3-1 1-1l-5-2-2-1h5l5 1-1 6c0 4 0 7 1 6l1-1 3-3c2-4 3-2 0 4-2 5-4 7-6 7s-2 0-1 1c0 1 2 1 6-1l5-2 3 1c2-1 3 0 3 1l1 3 1-1 2-2 1 1v1c2 0 6-5 6-7h2v-1l-1-2h1c2 0 2 0 1-1-1-2-1-2 1-2 1 1 2 1 2-1l4-2c2 0 2 0 1-1-2-2-2-3 0-3l1-2v-2l1-3 2-1c1 1 1 1 2-1l1-2 1-3c0-2 0-2 2-2 1 1 1 0 1-3l-1-4c-2 0-2 0 0-2l1-3-1-18V38h-3l-3 1h-5c-1 2-1 2-2 1l-3-1h-3c0 1-3 1-3-1l-2-1-1-1-1-2v2c-1 2-2 2-3 1h-2l-2 1c-2 0-1-2 1-2l1-1c-1-1-5 0-5 2-1 2-3 1-2-1 0-3-1-3-4 0-2 1-2 2-1 3v2l1 4 1 7 1 4 1 2-1 2v2c2 2 2 17 1 17l-1 1c-2 11-3 13-2 14l-1 2c-2 1-12 1-11-1l-6-2c-8-2-15-1-15 2l-2 1c-3 0-11-3-11-4 0-4 0-4-3-3l-5-1-6-3-4-1-4-1c-3 0-4 0-3-1s1-1-1-1v-1c2 0 2-3 1-5-1-1-2-3-1-4l-1-3c-2-4-2-4 0-4l1-1 1-3c1-1 2-2 1-6v-6h3c3 0 5-3 3-6l1-2 1-2c0-3 3-4 10-3h8l6 1 8 1c4 0 6 0 5-1l-3-1h-2 8c3 1 4 0 2-1l-1-3-2-4-2-3h-4c-6 0-6 0-7-4l-2-3v-1c2-2 3-3 1-3h-3l-6 1c-1 2-6 2-6 0 0-1-1-1-2 1l-4 2c-1-1-1 0-2 1l-1 1c0-2-7 3-7 5h-2l-5 2-4 4a59 59 0 0 0-17 47c1 3 1 3-1 3-5-2-6 2-5 13 1 9 6 19 8 19l1 13c0 14-2 20-9 29-4 5-13 11-16 11l-13 6h-2l-1-55-1-55a165 165 0 0 0-2 43l-1 7h-4c0 2-4 5-6 5H7V96L6 38 5 10l2-1c4-2 4-2-2-4L0 3l1 12m74 86c0 4 1 5 4 6 5 3 17 5 21 4l2-1v-7l-1-4c0-3-2-4-4-1l-4 1-2 2c0 4-6 3-11-2-5-4-5-4-5 2m2 9l2 1 1 2 1 3 1 2-1 2c-2 0 1 18 3 19l1 2c0 2 1 3 2 3l1 2c0 2 5 3 5 0v-4c-1-3-1-4 2-4v-2c5 0 6-4 5-11v-6l1-3-1-2-6-1-12-2-5-1m65 43a382 382 0 0 0-3 42l-1 5h21c12 0 20 0 19-1l-4-1-5-1-4-2-1-1v2l-2-1c-3-2-2-3 0-3l2-1h3l2-1 17 3c6 0 7 1 10 3 3 3 5 3 2 0v-3c2-1 2-2 1-2l-1-1 1-1c1 1 1-6 1-18l-1-19-4 2c-7 3-8 3-11 1-4-2-7-1-7 1l-1-1h-13c-1-1-1-1-1 1l-2 2-1-2c1-2 0-2-11-3l-5-1c-1-1-1 0-1 1m-23 25l-4 13c-4 10-5 9 9 9s13 1 11-9c-2-8-6-15-9-15l-2-1c-2-3-3-2-5 3' fill='lightgray' fill-rule='evenodd'/%3e%3c/svg%3e"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "transitionDelay": "0.25s",
                    "width": "100%",
                  }
                }
              />
              <picture>
                <source
                  srcSet="/static/ceeab084c21596435216ed7397fb861b/511b7/profile.jpg 1x,
/static/ceeab084c21596435216ed7397fb861b/5c0c2/profile.jpg 1.5x,
/static/ceeab084c21596435216ed7397fb861b/0fa17/profile.jpg 2x,
/static/ceeab084c21596435216ed7397fb861b/70cc3/profile.jpg 3x"
                />
                <img
                  alt="Profile Picture of Marcel Michau"
                  height={200}
                  onLoad={[Function]}
                  src="/static/ceeab084c21596435216ed7397fb861b/511b7/profile.jpg"
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 0.5s",
                      "width": "100%",
                    }
                  }
                  width={200}
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcSet=\\"/static/ceeab084c21596435216ed7397fb861b/511b7/profile.jpg 1x,
/static/ceeab084c21596435216ed7397fb861b/5c0c2/profile.jpg 1.5x,
/static/ceeab084c21596435216ed7397fb861b/0fa17/profile.jpg 2x,
/static/ceeab084c21596435216ed7397fb861b/70cc3/profile.jpg 3x\\" /><img width=\\"200\\" height=\\"200\\" src=\\"/static/ceeab084c21596435216ed7397fb861b/511b7/profile.jpg\\" alt=\\"Profile Picture of Marcel Michau\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
            <p>
              I write code & stuff
            </p>
          </div>
        </div>
      </header>
      <main
        className="sc-htoDjs janYlF sc-bxivhb uaYTM"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>This is my website. No one else's.</p>
<p>Look how minimal it is.</p>
<p>It is written using:</p>
<ul>
<li><a href=\\"https://reactjs.org/\\">React</a></li>
</ul>
<p>With:</p>
<ul>
<li><a href=\\"https://www.gatsbyjs.org/\\">Gatsby</a></li>
</ul>
<p>Using:</p>
<ul>
<li><a href=\\"https://www.styled-components.com/\\">Styled Components</a></li>
<li><a href=\\"https://fontawesome.com/\\">Font Awesome</a></li>
</ul>
<p>With some help from:</p>
<ul>
<li><a href=\\"https://prettier.io/\\">Prettier</a></li>
</ul>
<p>Hosted on:</p>
<ul>
<li><a href=\\"https://www.netlify.com/\\">Netlify</a></li>
</ul>
<p>It is intended to be:</p>
<ul>
<li>Fast 🐌</li>
<li>Agile 🐢</li>
<li>Scalable ⚖️</li>
<li>Cloud-native 🌧️</li>
<li>Enterprise-ready 🏢</li>
<li>Containerized 📦</li>
<li>Microservice-based 🔍</li>
<li>Powered by Machine Learning 🤖</li>
<li>Blockchain ₿</li>
<li>Filled with all the buzz words so that search engines pick this up 😁</li>
</ul>
<p>This is what I'm going to use it for:</p>
<ul>
<li>To showcase some of the things I'm working/have worked on ✨</li>
<li>To play around &#x26; experiment with cool new tech 🎮</li>
<li>To have some slice of the internet which is my own 🍰</li>
<li>World domination 🤘</li>
</ul>
<p>This is what I might use it for:</p>
<ul>
<li>To write blogs ✏️</li>
<li>To list some of the things I'm good at 🖥️</li>
</ul>
<p>This is what I'm not going to use it for:</p>
<ul>
<li>Sell stuff 💰</li>
<li>Evil 💀</li>
</ul>",
            }
          }
        />
      </main>
      <footer
        className="sc-dnqmqq nlIOq sc-ifAKCX fVzfsA"
      >
        <div>
          <p>
            Marcel Michau © 
            2019
          </p>
          <div
            className="sc-EHOje lkTEkl"
          >
            <a
              href="https://github.com/MarcelMichau"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-github fa-w-16 fa-lg "
                data-icon="github"
                data-prefix="fab"
                role="img"
                style={Object {}}
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
            <a
              href="https://twitter.com/MarcelMichau"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter fa-w-16 fa-lg "
                data-icon="twitter"
                data-prefix="fab"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
